### PR TITLE
[sival, i2c] Remove the speed check from the testplan

### PR DESCRIPTION
**Pull request by engdoreis**
Checking the baudrate accuracy in the automatic test would be quite challenging, so I'm proposing in this PR that we change the test plan to check the throughput and let the baudrate to be tested manually using external measurement equipment.

### CODE REVIEW

**Review by a-will — APPROVED** [review]
This seems fine to me, but we may want some kind of characterization of performance during validation. I'm not sure we know what are reasonable values yet, though.

File: hw/top_earlgrey/data/ip/chip_i2c_testplan.hjson [hunk 1]
```diff
@@ -53,23 +53,19 @@
     }
     {
       name: chip_sw_i2c_speed
-      desc: '''Verify the transmission speed of data over the chip's I2C interfaces.
+      desc: '''Verify the correctness data transmission over the chip's I2C interfaces with different speeds.
```
Comment (matutem) [inline hw/top_earlgrey/data/ip/chip_i2c_testplan.hjson:56]: Change to either "...the correctness of data..." or "Verify data transmission is correct over..."

**Comment by matutem** [comment]
I am curious why this measurement is challenging: there are various timers in the device (like the rv_timer), and you can also measure at the host. You could do a long transmission to minimize the start/stop overhead, like initializing a chunk of SRAM and pump it out? I may be missing something.

**Comment by a-will** [comment]
> I am curious why this measurement is challenging: there are various timers in the device (like the rv_timer), and you can also measure at the host. You could do a long transmission to minimize the start/stop overhead, like initializing a chunk of SRAM and pump it out? I may be missing something.

For a certain measurement, I think you're right, but it really depends on what we want to test. If we want generic throughput, we'd need either a reasonable target spec or a guess of a pretty wide range of acceptable values, but we could measure transactions based on the time between submission and the done IRQ.

If we want to know what the baud rate is (which is what the text initially said here), that's a pretty specific thing that requires more precise measurement timing + an understanding that the nominal values are _not_ the correct targets (which we got wrong on several iterations of DV).

If we're going for throughput characterization, then I think we should make the text say that, not claim we're checking the baud rate (SCL toggle rate).

**Comment by engdoreis** [comment]
Yes, I agree that throughput is fairly easy to measure. Should I change the test plan to check throughput instead?

**Comment by a-will** [comment]
> Yes, I agree that throughput is fairly easy to measure. Should I change the test plan to check throughput instead?

That sounds good to me!

**Comment by engdoreis** [comment]
I changed the test plan to verify the throughput.

**Review by a-will — APPROVED** [review]
LGTM. Just be prepared to be surprised by what the "appropriate throughput" is, starting with what baud rate you actually get after satisfying all timing constraints, haha.

**Pull request merged**